Intercoms are now louder

atlantiscze opened the pull request:
- Intercoms have been brought on par with holopads, meaning they have up to 7 tiles range (which is the view range as well, so when you see an intercom which is turned on, you'll hear it)
- Right now intercoms are pretty useless (especially since holocalls were added) unless you sit literally next to one. Hopefully they will be more useful now.

:cl:
tweak: Intercoms now have better speaker and microphone range.
tweak: Intercoms now have subtle status overlays for microphone and speaker, allowing to visually tell whether intercom/speaker is enabled or not.
/:cl:

atlantiscze committed:
Intercoms are now louder
- Intercoms have been brought on par with holopads, meaning they have up to 7 tiles range (which is the view range as well, so when you see an intercom which is turned on, you'll hear it)
- Right now intercoms are pretty useless (especially since holocalls were added) unless you sit literally next to one. Hopefully they will be more useful now.

chaoko99 commented:
Might want to change that to speaking range. Hearing implies the microphone can be activated from a range.

atlantiscze commented:
Both speaker and microphone are tied to the same variable. Amended the changelog entry to be more specific.

Kelenius commented:
>Stun someone
>Strip their headset
>They yell into an intercom in a different room

atlantiscze commented:
Kelenius, that implies the intercom's microphone has been turned on. They are all off by default. And of course as all other radios, it does respect sight blocking obstacles such as walls.

Kelenius commented:
Well, replace that with "on the other side of the room", and most rooms on the station have one.

As for microphone, I am fairly certain `:i` still works.

Additionally, it upgrades AI from ValidHunter 2000 to ValidHunter Supreme, because they can now spy on practically the entire station's conversations.

atlantiscze commented:
I believe !i only works on adjacent intercom, though if not, it can be easily tweaked.

AIs abusing the feature:
I main AI so i know this area pretty well. Right now they are useless even when you are traitor/malf, as you can't usually hear the whole conversation (as most of the cases at least one person is out of the absurdly low 2 tile range). However setting all intercoms to microphone on is both very annoying to do as its a lot of clicking with not really good gain, and most importantly, it is very likely to get noticed everntually, causing iC confrontation with the AI.

Overall, as AI main player, i consider intercom spying to be an antag only tool, or a thing that can be used when explicitly asked for, for example by security. I very rarely use it outside of these circumstances, and when i do its usually a very specific thing where i use only a single intercom for short duration, so nothing on the scale you are describing, as that's simply not practical (all it needs is single person finding a single intercom you modified and you have a problem)

I would also like to point out that if you believe someone is abusing the mechanic OOCly (such as AI turning on all intercoms without being antag or having an order) you can adminhelp it. I don't think we should block people which want to use intercoms in legit way from doing so, on the grounds of "Someone may abuse it", when those few who might abuse it can be handled administratively.

Kelenius commented:
>I believe !i only works on adjacent intercom, though if not, it can be easily tweaked.

I think it works anywhere in the range, but I might be wrong.

>Right now they are useless even when you are traitor/malf, as you can't usually hear the whole conversation

And you shouldn't.

>However setting all intercoms to microphone on is both very annoying to do as its a lot of clicking

Click on intercom, speaker off, frequency to AI private, mic on. Four clicks per intercom.

>it is very likely to get noticed everntually, causing iC confrontation with the AI

By which point you have already heard the conversation. Also, not an issue if you're spying on antags.

>Overall, as AI main player, i consider intercom spying to be an antag only tool, or a thing that can be used when explicitly asked for, for example by security

Can you vouch for all other AI players?

>I would also like to point out that if you believe someone is abusing the mechanic OOCly

Where exactly is the abuse here, if the intercoms are changed to work like this? It's going to be a legitimate use of game mechanics working exactly as intended: if intercoms can pick up conversations in a 7 tile range, why wouldn't AI use that? How do you imagine that ahelp? "AI is spying on me?"

>I don't think we should block people which want to use intercoms in legit way from doing so, on the grounds of "Someone may abuse it"

Brb adding a mininuke to CO's room that immediately blows the ship when you press a button. It has legit uses, so I don't think we should consider people who might abuse it.

Sbotkin commented:
This conversation is useless as the shitstorm is going on the forum by salty antags.

Kelenius commented:
>This conversation is useless as the shitstorm is going on the forum by salty AI players.

FTFY.

SparklySheep commented:
How close do you need to be to the intercom to have :i work with this change? Has that been unchanged?

If you don't know, that should be tested.

atlantiscze commented:
This particular thing was not tested, to be honest i didn't even know about :i before people mentioned it. I'll see if i can test this locally, though i am limited to my craptop at this time, so it might take some time.

Crushtoe commented:
I'm against it personally but if it's really controversial make it an experiment and if it doesn't go stellar within five days unmerge it.

atlantiscze commented:
A compromise i am willing to do in this particular PR is some subtle visual way to indicate that intercom's speaker/microphone is turned on (probably a simple one-pixel LED overlay). Unfortunately my coding capacities are somewhat limited as my main computer is currently unavailable.

atlantiscze committed:
Status LEDs
- Intercoms now have subtle two-pixel status lights for speaker and microphone status. These replace stationary lights that existed on the sprite even before - they are now just tied to the microphone/speaker status.

atlantiscze commented:
Updated, added subtle visual overlays.

Also as a note, i have tested :i locally, and it does not seem to work at all, no matter how close or far you are from the intercom.

Vivalas commented:
I can see increasing the speaker range but microphone range really shouldn't be that far, maybe halfway across the room. People aren't constantly yelling after all. I hate to jump on the antag bandwagon but this would be a pretty large nerf to antags.

comma commented:
👎 
Intercomm spying is annoying enough without making it even easier.
There's close to none legitimate uses for intercomms when you have headsets, and they can all be accomplished by being near it or getting a handheld.

atlantiscze commented:
There are many legitimate uses. Typical situation just from few rounds ago:

Charon was outfitted with relay, and expedition team used frequency 145.7 for their local comms. We (bridge crew) wanted to listen to these comms, though intercoms had range of mere two tiles or so, making them useless for this task. Unless you are incredibly lucky to be sitting on top of one, you have to use a portable radio.

Another legitimate use example:
Emergency communications. Imagine telecomms gets blown up. Right now, if you don't have a portable radio, you're screwed, period.  With this change intercoms become a usable alternative, and you will be able to, for example, set up intercom on wall in the room you are working in, and continue your duties. 

And so on. These are all legitimate and useful applications.
As intercom spying was brought to my attention, i was was willing to compromise, and added little status lights. If you are careful, and check the intercom visually, you'll be able to notice that its transmitting your possibly confidential conversation.

**I would be willing for one last compromise** - a manual "main power" on/off switch in the intercom's UI. Icon of turned off intercom is pretty different from turned on one, so this would invalidate the "But it is easy to miss" argument, however you would have to turn off the intercom manually.

Kelenius commented:
>With this change intercoms become a usable alternative, and you will be able to, for example, set up intercom on wall in the room you are working in, and continue your duties.

Your radio headset can work as a _receiver_ when the telecomms are out _just fine_. If you want to speak, go and walk to the nearest intercom, say what you want, and continue on your merry way. Or get a handheld like a baby. They're all over the ship.

Hubblenaut commented:
~~I'm confused about the speaker range, I'm pretty sure you already *can* listen to an intercom that is anywhere in your range of view.~~

Nevermind, it cannot. Fully supporting this.

Also :+1: for the status lights.

Vivalas commented:
None of those legitimate examples requires increasing the range of intercoms so much though. In the event of a comms outtage you know what the new meta will be? Turning on all intercoms in your workplace so you can scream for help just as easily as if you had a headset working.

atlantiscze commented:
Judging from the thumbs up/down on the PR, people are mostly neutral/slightly in favor (though the "against" group is incredibly more vocal) of the change. For now i consider this PR feature wise complete. Balance can further be tweaked if necessary.

Hubblenaut commented:
> None of those legitimate examples requires increasing the range of intercoms so much though. In the event of a comms outtage you know what the new meta will be? Turning on all intercoms in your workplace so you can scream for help just as easily as if you had a headset working.

Ooor you just have one of the dozen station-bounced radios on you.

Chiquiii commented:
Confliicts

PurpleMartinJCK commented:
You've got a couple days to get a Dev to support this or it will be closed

ghost commented:
@comma @PsiOmegaDelta @Techhead0 See above.

chaoko99 commented:
I don't think it's fair to punish the submitter because the devs are playing hookey.

atlantiscze committed:
Merge branch 'dev' into 2017-12-26-LouderIntercoms

atlantiscze commented:
Tried to fix conflicts through github editor as i still don't have access to a proper one. Hopefully nothing broke.

comma commented:
>devs are playing hookey

If someone agreed this is a good addition, they had two months to merge it. Don't spin it pls.

chaoko99 commented:
Then don't go silent.

ghost commented:
>If someone agreed this is a good addition, they had two months to merge it. Don't spin it pls.

Then... Why not close it if it isn't? :confused:

comma commented:
You know, good point

atlantiscze commented:
@comma 
The PR was in conflicting state for quite some time, as i didn't have time to resolve that conflict due to some higher priority stuff happening IRL. I admit this was my fault, but right now, the PR has been in functional state for a day or two and it is ready for review/merge again.

There is a total of +11/-6 thumbs up/down ratio, which indicates that this change is relatively well supported (at least among people here on github). I've also gone through few balancing compromises based on feedback earlier.

So i believe your decision to close this was kind of harsh, and i'd like to request this to be reopened. If you believe it is contentious in any way, it is a matter that can be handled on the game changes forum.
